Adding architecture guidelines to Red Hat Style guide (#295)
redhat-documentation/supplementary-style-guide: supplementary_style_guide/glossary_terms_conventions/general_conventions/a.adoc

[discrete]
[[AArch64]]
==== image:images/yes.png[yes] AArch64 (noun)

> *Description*: 64-bit version of the ARM architecture. Use this term format when referring to OS and server instances. Including: Linux versions, RHEL, or CoreOS. Use uppercase format in general sentences. 

Reviewer: A 64-bit...

I know, `64-bit` is acting as a noun, but starting with a character reads better than starting a sentence with a number. 

--------------
"Use uppercase format in general sentences." 

Not sure what this applies to? CoreOS would remain as is: a mixture of lowercase or uppercase.

Example:

> _If you are using RHEL with an AArch64 system, run the following command_

Reviewer: _If you are using RHEL with an AArch64 system, run the following command_

Consider following the style adopted throughout the SSG. Example: 

> For example, "When running the JBoss EAP server in a managed domain".

*Use it*: yes

*Incorrect forms*:

*See also*:

[discrete]
[[aarch64]]
==== image:images/yes.png[yes] aarch64 (noun)

> *Description*: 64-bit version of the ARM architecture. Use this term format when referring to OS and server instances. Including: Linux versions, RHEL, or CoreOS. Use lowercase (and backticks) when referring to objects or parameters. 

Reviewer: Term format > term ?

[discrete]
[[AArch64]]
==== image:images/yes.png[yes] AArch64 (noun)

> *Description*: A 64-bit version of the ARM architecture. Use this term when referring to Operating systems and server instances. Including: Linux versions, RHEL, or CoreOS. Use this format in general cases when refering to system architecture. 

Reviewer: Operating systems > operating systems

Cloud providers can use different formats of this term when using architectures. If you are documenting hard code, commands or outputs, confer with your SME on the correct format for the specific use case.

Example:

> _When running RHEL with an AArch64 system, run the following command._

Reviewer: _When running RHEL with an AArch64 system, run the following command._ > _When running RHEL with an AArch64 system, run the following command:_

*Use it*: yes

*Incorrect forms*:

> *See also*: xref:aarch64[aarch64]

Reviewer: *See also*: xref:aarch64[_aarch64]

Author: Ah! Thank you, my apologies, I keep missing these small things

[discrete]
[[aarch64]]
==== image:images/yes.png[yes] aarch64 (noun)

> *Description*: A 64-bit version of the ARM architecture. Use this term when referring to Operating systems and server instances. Including: Linux versions, RHEL, or CoreOS. Use this format (and backticks) when referring to objects or parameters. 

Reviewer: Use this format (and backticks) when referring to objects or parameters.

> Surround the term with backticks when referring to objects or parameters.

Cloud providers can use different formats of this term when using architectures. If you are documenting hard code, commands, or outputs, confer with your SME on the correct format for the specific use case.

Example:

_Specify the system architecture of your cluster, such as `x86_64` or `aarch64`._

*Use it*: yes

*Incorrect forms*:

> *See also*: xref:AArch64[AArch64]

Reviewer 2: This cross-reference link is still not working because the system sees it as identical to the `[aarch64]` link above.

Reviewer: *See also*: xref:AArch64[AArch64]

> *See also*: xref:AArch64[aarch64]

?

Reviewer 2: You're right. This one is working for me, too. Setting to Resolved.

> *See also*: xref:AArch64[AArch64]

Reviewer: This link does not work for me (could be something I my side, but I think I have the most up-to-date version of your branch). Shoudl this state:

*See also*: xref:AArch64[aarch64]

?

[discrete]
[[AMD64]]
==== image:images/yes.png[yes] AMD64 (noun)

> *Description*: AMD's 64-bit version of the x86 architecture. Use this format for OpenShift Container Platform attributes, Kubernetes, Operators, APIs, CLI objects. Use this format in general sentences when referring to OCP features. 

Reviewer: AMD's 64-bit version of the x86 architecture. Use this format for OpenShift Container Platform (OCP) attributes, Kubernetes, Operators, APIs, CLI objects, and so on. Use this format in general sentences when referring to OCP features.

> *Description*: AMD's 64-bit version of the x86 architecture. Use this term for OpenShift Container Platform (OCP) attributes, Kubernetes, Operators, APIs, or CLI objects. Use this format in general sentences when referring to OCP features. 

Reviewer: Sorry, possessive here also :grimacing:

Author: All good! Good catch!

Cloud providers can use different formats of this term when using architectures. If you are documenting hard code, commands, or outputs, confer with your SME on the correct format for the specific use case.

Example: 

_This Operator is supported on AMD64 and ARM64 platforms._ 

*Use it*: yes

*Incorrect forms*: _AMD64_ 

*See also*: xref:amd64[amd64]

[discrete]
[[amd64]]
==== image:images/yes.png[yes] amd64 (noun)

> *Description*: AMD's 64-bit version of the x86 architecture. Use this term for OpenShift Container Platform attributes, Kubernetes, Operators, APIs, or CLI objects. Use the lowercase format and backticks when referring to objects or parameters.

Reviewer: Sorry, possessive here also :grimacing:

> *Description*: AMD's 64-bit version of the x86 architecture. Use this term for OpenShift Container Platform attributes, Kubernetes, Operators, APIs, CLI objects. Use lowercase (and backticks) when referring to objects or parameters.

Reviewer 3: ```suggestion
*Description*: AMD's 64-bit version of the x86 architecture. Use this term for OpenShift Container Platform attributes, Kubernetes, Operators, APIs, and CLI objects. Use lowercase (and backticks) when referring to objects or parameters.
```

Reviewer 4: Should be _Operators_ capitalized?

And I would remove the brackets around _and backticks_.

Author: Yes we capitalize Operators in OCP docs

Cloud providers can use different formats of this term when using architectures. If you are documenting hard code, commands, or outputs, confer with your SME on the correct format for the specific use case.

Example:

_Valid values are `amd64`._

*Use it*: yes

*Incorrect forms*:

> *See also*: xref:AMD64[AMD64]

Reviewer 2: This cross-reference link is still not working because the system sees it as identical to the [amd64] link above.

Reviewer: Ah, yes, well spotted. 

See also: [AArch64](https://github.com/redhat-documentation/supplementary-style-guide/pull/295#AArch64) does not work

See also: [aarch64](https://github.com/redhat-documentation/supplementary-style-guide/pull/295#aarch64) points to Arch64 (noun)

Reviewer 2: Yes. The problem is case-insensitivity.

Reviewer: This hyperlink works for me.

Reviewer 2: My mistake. Working for me, too. Setting to resolved.

==== image:images/yes.png[yes] ARM64 (noun)
*Description*: A 64-bit version of the ARM architecture. Use this term for OpenShift Container Platform (OCP) attributes, Kubernetes, Operators, APIs, and CLI objects. Use this format in general sentences when referring to OCP features.

Cloud providers can use different formats of this term when using architectures. If you are documenting hard code, commands, or outputs, confer with your SME on the correct format for the specific use case.

Example: 

_Creating an ARM64 compute machine set._ 

*Use it*: yes

*Incorrect forms*: 

*See also*: xref:arm64[arm64]

[discrete]
[[arm64]]
==== image:images/yes.png[yes] arm64 (noun)
*Description*: A 64-bit version of the ARM architecture. Use this term for OpenShift Container Platform attributes, Kubernetes, Operators, APIs, and CLI objects. Use lowercase format and backticks when referring to objects or parameters.

Cloud providers can use different formats of this term when using architectures. If you are documenting hard code, commands, or outputs, confer with your SME on the correct format for the specific use case.

Example: 

_Valid values are `arm64`._

*Use it*: yes

*Incorrect forms*: 

> *See also*: c

Reviewer 2: ```suggestion
*See also*: c
```
Replace `c` with a cross-ref to the ARM64 entry

[discrete]

> [[64-bit ARM]]

Author: I noticed that the 64-bit ARM and 64-bit x86 entries are not rendering properly in my local preview. I'm not sure if this will be resolved after merge but I wanted to ask someone here s well

> [[64-bit ARM]]

Reviewer 5: ```suggestion
[[64-bit-arm]]
```
The anchor in the brackets should be lowercase and hyphenated in place of spaces. Ditto other entries.

Reviewer: Yes, @kelbrown20 . I do not see why not. You could describe the usage of the term by surrounding it in quotation marks. 

For example:

Use the term "64-bit ARM".

Reviewer: As an example, see "Kubernetes master" in the [SSG](https://redhat-documentation.github.io/supplementary-style-guide/#_m). Just one example of many, but might go into a bit more detail about usage, but gives some sort of example.

Author: Oh I see how it works! Thank you!

> [[bit-64-arm]]

Author: Hmm, anytime I tried to start it with the number, it was still not rendering properly. Does this look okay? Its not ideal, but wasn't sure how to get it to look correct with the number in front

Reviewer 5: @kelbrown20 It looks ok to me in the GitHub preview.

Reviewer: ID tags are hidden and are only visible to a doc user through the URL or an inspection tool, so this should not be an issue.

Author: Gotcha! Thank you!

==== image:images/yes.png[yes] 64-bit ARM (noun)

> *Description*: A 64-bit version of the ARM architecture. This term can refer to both AArch66/`aarch64` and ARM64/`arm64`. Use this format in general cases when writing for both names of the architecture for various cloud providers.

Reviewer 2: ```suggestion
*Description*: A 64-bit version of the ARM architecture. This term can refer to both AArch66/`aarch64` and ARM64/`arm64`. Use this format in general cases when writing both for names of the architecture and for various cloud providers.
```

> *Description*: Use in general cases when writing for both names of the architectures for various cloud providers. The term 64-bit ARM systems can refer to AArch64 and ARM64/`arm64`. Using this format can avoid confusion for users when the writer wants to write a general description of the architectures. 

Reviewer 5: ```suggestion
*Description*: A 64-bit version of the ARM architecture. This term can refer to both AArch64 and ARM64/`arm64`. Use this format in general cases when writing for both names of the architectures for various cloud providers. 
```
Does the "using this format" section add any additional value or describe a new use case? If the latter, incorporate into the "Use this format in general cases..." sentence.

Author: Its mostly to introduce the term, but I like your suggestion. Thank you!
